components/gcp-click-to-deploy/src/DeployForm.tsx

```diff
@@ -380,17 +393,68 @@ export default class DeployForm extends React.Component<any, DeployFormState> {
       (error, response, body) => {
         if (error) {
           this._appendLine('Could not reach backend server, exiting');
+          this.setState({
+            dialogTitle: 'Could not reach backend server, exiting',
+          });
           return;
         }
       }
     );
+    if (this.state.dialogTitle) {
+      return;
+    }
+
+    const email = await Gapi.getSignedInEmail();
+
+    // Create service account in target project, and use this sa's tmp token for all following actions.
+    const projectNumber = await Gapi.cloudresourcemanager.getProjectNumber(project)
+      .catch(e => {
+        this.setState({
+          dialogBody: 'Error trying to get the project number: ' + e,
+          dialogTitle: 'Deployment Error',
+        });
+        return undefined;
+      });
+
+    if (this.state.dialogTitle) {
+      return;
+    }
+
+    this._appendLine('Proceeding with project number: ' + projectNumber);
+
+    const deploymentName = this.state.deploymentName;
+    const accountId = deploymentName + '-owner';
+    const saEmail = accountId + '@' + project +'.iam.gserviceaccount.com';
+    const saExists = await Gapi.iam.serviceAccountExist(project, saEmail);
+    if (!saExists) {
+      await Gapi.iam.createServiceAccount(project, accountId);
+    }
+
+    const currProjPolicy = await Gapi.cloudresourcemanager.getIamPolicy(project);
+    const bindingKey = 'bindings';
+    currProjPolicy[bindingKey].push({
+      'members': ['serviceAccount:' + saEmail],
+      'role': 'roles/owner'
+    });
+    await Gapi.cloudresourcemanager.setIamPolicy(project, currProjPolicy);
```

Comment:
We should probably add a `.catch` here to use the dialog to show the error properly, since we don't have a try/catch in parent scope. Same for all service calls, at least the ones that we think can fail, such as this one, and the `setServiceAccountIAM` call below.